components/textdiff/textdiff.js

```diff
@@ -198,7 +208,7 @@ class TextDiffViewModel {
     }
     return `<div class="d2h-code-line-prefix"><span data-bind="visible: editState() !== 'patched'">${symbol}</span><input ${
       isActive ? 'checked' : ''
-    } type="checkbox" data-bind="visible: editState() === 'patched', click: togglePatchLine.bind($data, ${index})"></input></div>`;
+    } type="checkbox" data-bind="visible: editState() === 'patched', click: togglePatchLine.bind($data, ${index}), clickBubble: false"></input></div>`;
```

Comment:
Since the checkbox is inside the row, this click event on the checkbox is not really needed, right?

Reply:
Well, it's needed in the sense that it's invoked here:
https://github.com/FredrikNoren/ungit/blob/406a4b0652a9592d2899895019625b7100c3fdf2/components/textdiff/textdiff.js#L228-L231
and simply removing it means the functionality no longer works :)

But yes, I agree it feels like it should be redundant (I originally made these changes as a quick-and-dirty hack for my own use some months ago, but the conversation at #1421 prompted my to PR them).

The alternative, it seems to me, is to bind the "checked" state of the checkbox to the index in `TextDiffViewModel.patchLineList` and binding `togglePatchLine(index)` to the click event on the `<td/>` -- so I've added a commit that does that.

There are two downsides to this: 1) is that, as the comment at https://github.com/FredrikNoren/ungit/blob/406a4b0652a9592d2899895019625b7100c3fdf2/components/textdiff/textdiff.js#L168-L170 mentions, it is necessary to force a re-render; and 2) is that the `data-bind="checked: ...` binding is two-way, so the normal behaviour of the checkbox has to be inhibited otherwise a) the checkbox is changed and the immediately re-changed, and b) `togglePatchLine(index)` is only called once, and the interface ends up in an inconsistent state.  I've addressed that by simply applying `point-events: none` to the checkbox itself (making the `checked` binding effectively one-way), but there may be a better way?

All in all, I'm not 100% convinced this is better than before -- what do you think?